mm/kfence/core.c

```diff
@@ -191,10 +191,13 @@ static noinline void metadata_update_state(struct kfence_metadata *meta,
 
 	lockdep_assert_held(&meta->lock);
 
-	if (next == KFENCE_OBJECT_FREED)
+	if (next == KFENCE_OBJECT_FREED) {
 		meta->num_free_stack = nentries;
-	else
+		meta->free_task_pid = task_pid_nr(current);
```

Comment:
Maybe it's time to factor alloc/free metadata into a separate struct?

Reply:
Ah yes of course, done!